arch/arc/core/fault_s.S

```diff
@@ -112,29 +116,28 @@ _exc_return:
 	ld_s r2, [r1, _kernel_offset_to_ready_q_cache]
 	st_s r2, [r1, _kernel_offset_to_current]
 
-#ifdef CONFIG_ARC_HAS_SECURE
-	/*
-	 * sync up the ERSEC_STAT.ERM and SEC_STAT.IRM.
-	 * use a fake interrupt return to simulate an exception turn.
-	 * ERM and IRM record which mode the cpu should return, 1: secure
-	 * 0: normal
-	 */
-	lr r3,[_ARC_V2_ERSEC_STAT]
-	btst r3, 31
-	bset.nz r3, r3, 3
-	bclr.z r3, r3, 3
-	/* sflag r3 */
-	/* sflag instruction is not supported in current ARC GNU */
-	.long 0x00ff302f
-#endif
 	/* clear AE bit to forget this was an exception */
 	lr r3, [_ARC_V2_STATUS32]
 	and r3,r3,(~_ARC_V2_STATUS32_AE)
 	kflag r3
-	/* pretend lowest priority interrupt happened to use common handler */
-	lr r3, [_ARC_V2_AUX_IRQ_ACT]
-	or r3,r3,(1<<(CONFIG_NUM_IRQ_PRIO_LEVELS-1)) /* use lowest */
+
+	mov r3, 0
+	st r3, [exc_nest_count]
+	/* pretend lowest priority interrupt happened to use common handler
```

Comment:
Since this requires a sjli call in Normal mode, maybe better now to not use re-use the common handler?
Guess this code is not in a timing critical path, but avoiding a slji would be good. It's a powerfull/dangerous sjli call so should not be use-able by other normal code (implementation of the sjli needs to check the caller)